x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/execution/search/OrdinalTests.java

```diff
@@ -83,51 +85,74 @@ public void testCompareToSameTsSameTieNullDifferentImplicitTb() {
+        assertFalse(new Ordinal(randomTimestampBetween(0, 999), null, randomLong()).between(before, after));
+        assertFalse(new Ordinal(randomTimestampBetween(7000, 8000), null, randomLong()).between(before, after));
     }
 
     public void testTestBefore() {
-        Ordinal before = new Ordinal(randomLongBetween(1000, 2000), (Comparable) randomLong(), randomLong());
-        Ordinal after = new Ordinal(randomLongBetween(5000, 6000), (Comparable) randomLong(), randomLong());
+        Ordinal before = new Ordinal(randomTimestampBetween(1000, 2000), (Comparable) randomLong(), randomLong());
+        Ordinal after = new Ordinal(randomTimestampBetween(5000, 6000), (Comparable) randomLong(), randomLong());
 
         assertTrue(before.before(after));
         assertFalse(before.before(before));
         assertFalse(after.before(before));
     }
 
     public void testBeforeOrAt() {
-        Ordinal before = new Ordinal(randomLongBetween(1000, 2000), (Comparable) randomLong(), randomLong());
-        Ordinal after = new Ordinal(randomLongBetween(5000, 6000), (Comparable) randomLong(), randomLong());
+        Ordinal before = new Ordinal(randomTimestampBetween(1000, 2000), (Comparable) randomLong(), randomLong());
+        Ordinal after = new Ordinal(randomTimestampBetween(5000, 6000), (Comparable) randomLong(), randomLong());
 
         assertTrue(before.beforeOrAt(after));
         assertTrue(before.beforeOrAt(before));
         assertFalse(after.beforeOrAt(before));
     }
 
     public void testTestAfter() {
-        Ordinal before = new Ordinal(randomLongBetween(1000, 2000), (Comparable) randomLong(), randomLong());
-        Ordinal after = new Ordinal(randomLongBetween(5000, 6000), (Comparable) randomLong(), randomLong());
+        Ordinal before = new Ordinal(randomTimestampBetween(1000, 2000), (Comparable) randomLong(), randomLong());
+        Ordinal after = new Ordinal(randomTimestampBetween(5000, 6000), (Comparable) randomLong(), randomLong());
 
         assertTrue(after.after(before));
         assertFalse(after.after(after));
         assertFalse(before.after(after));
     }
 
     public void testAfterOrAt() {
-        Ordinal before = new Ordinal(randomLongBetween(1000, 2000), (Comparable) randomLong(), randomLong());
-        Ordinal after = new Ordinal(randomLongBetween(5000, 6000), (Comparable) randomLong(), randomLong());
+        Ordinal before = new Ordinal(randomTimestampBetween(1000, 2000), (Comparable) randomLong(), randomLong());
+        Ordinal after = new Ordinal(randomTimestampBetween(5000, 6000), (Comparable) randomLong(), randomLong());
 
         assertTrue(after.afterOrAt(before));
         assertTrue(after.afterOrAt(after));
         assertFalse(before.afterOrAt(after));
     }
+
+    static Object randomTimestamp() {
+        if (randomBoolean()) {
+            return randomLongBetween(-3155701416721920000L, 3155688986440319900L);
```

Comment:
where do these bounds come from? Will timestamps never be outside of these ranges?

Reply:
Those came from Instant, that has some (private) ranges for seconds. But this has been now improved.